File: Assignment09/Q4.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  MultDigits
//      Input:          integer
//      Output:         integer
//      Description:    Accept number from user and return multiplication
//                      of all digits
//      Date:           28/05/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int MultDigits(
                int iNo              // Input number
              )
{
    int iDigit = 0;                 // To store digit
    int iMult = 1;                  // To store multiplication result

    // Convert negative number into positive
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    // Traverse digits
    while(iNo > 0)
    {
        iDigit = iNo % 10;          // Extract last digit

        // Multiply digit
        iMult = iMult * iDigit;

        iNo = iNo / 10;             // Remove last digit
    }

    return iMult;                   // Return result
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to test MultDigits function
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;                 // Input number
    int iRet = 0;                   // Result

    printf("Enter number\n");

    scanf("%d", &iValue);

    // Function call
    iRet = MultDigits(iValue);

    // Output display
    printf("%d\n", iRet);

    return 0;                       // Successful termination
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      2395
//      Output:     270
//
//      Input:      1018
//      Output:     8
//
//      Input:      9440
//      Output:     144
//
//      Input:      922432
//      Output:     864
//
/////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////////////////////
//
//      Time Complexity Analysis
//
//      Let n be the number of digits in the input number
//
//      The while loop runs once for each digit
//      Each iteration performs constant time operations O(1)
//
//      Therefore,
//      Time Complexity = O(n)
//
//      Space Complexity = O(1)
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment19/Q4.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q4.java
//      Project   : Logic Building Assignment 19
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : DigitLogic
//      Method Name: printDigits
//      Input      : Integer
//      Output     : Void
//      Description: Print each digit of a number
//                   separately
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class DigitLogic
{
    void printDigits(int num)
    {
        int digit = 0;      // Stores extracted digit

        // Convert negative number to positive
        if(num < 0)
        {
            num = -num;
        }

        // Extract and display digits
        while(num != 0)
        {
            digit = num % 10;

            System.out.println(digit);

            num = num / 10;
        }
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q4
{
    public static void main(String args[])
    {
        // Create object of DigitLogic class
        DigitLogic obj = new DigitLogic();

        // Function call
        obj.printDigits(9876);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 9876
//
//      Output :
//      6
//      7
//      8
//      9
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
///////////////////////////////////////////////////////////////////
